Engine/gfx/blender.h

```diff
@@ -62,4 +62,109 @@ void set_opaque_alpha_blender();
 // Sets argb2argb for 32-bit mode, and provides appropriate funcs for blending 32-bit onto 15/16/24-bit destination
 void set_argb2any_blender();
 
+// ===============================
+// [AVD] Custom blenders for software BlendMode implementation
+// If we ditch software rendering we can remove this whole section
+#include <allegro.h>
+#include <allegro/internal/aintern.h>
+
+#define BLEND(bpp, r, g, b)   _blender_trans##bpp(makecol##bpp(r, g, b), y, n)
+// some formulas from AGSBlend
+#define _BLENDOP_BURN(B,L) ((B + L < 255) ? 0:(B + L - 255)) // note: burn was called subtract
+#define _BLENDOP_LIGHTEN(B,L) ((L > B) ? L:B)
+#define _BLENDOP_DARKEN(B,L) ((L > B) ? B:L)
+#define _BLENDOP_EXCLUSION(B,L) (B + L - 2 * B * L / 255)
+#define _BLENDOP_SUBTRACT(B,L) (L - B < 0) ? 0:(L - B)
+// not very pretty but forces the original alpha of "x" to "blender_result"
+inline unsigned long _blender_mask_alpha24(unsigned long blender_result, unsigned long x, unsigned long y, unsigned long n)
```

Comment:
This is not critical, but I'd like to note that defining these functions inlined in the header makes little sense, because they are passed as pointers to sprite drawing routine, and code inlining will not happen anyway.